### PR TITLE
Add delete card func

### LINKED ISSUE #1957: Allow deletion of LL cards

### PR DESCRIPTION
**Pull request by bramkragten**
Closes: https://github.com/home-assistant/home-assistant-polymer/issues/1957

### CODE REVIEW

File: src/panels/lovelace/components/hui-card-options.ts [hunk 1]
```diff
@@ -52,17 +54,32 @@ export class HuiCardOptions extends LitElement {
...
       cardConfig: this.cardConfig!,
       reloadLovelace: () => fireEvent(this, "config-refresh"),
     });
   }
+  private _deleteCard(): void {
+    confDeleteCard(
+      this.hass!,
+      this.cardConfig!,
+      () => fireEvent(this, "config-refresh"),
+      () => this._editCard()
```
Comment (bramkragten) [inline src/panels/lovelace/components/hui-card-options.ts]: This makes no sense, can better check for an ID before calling delete

File: src/panels/lovelace/common/data.ts [hunk 2]
```diff
@@ -38,3 +38,12 @@ export const updateCardConfig = (
     card_config: config,
     format: configFormat,
   });
+
+export const deleteCard = (
```
Comment (balloob) [inline src/panels/lovelace/common/data.ts]: Can we do an inbetween PR to move this file to `src/data/lovelace.ts` (like we have for other components)

File: src/panels/lovelace/common/data.ts [hunk 3]
```diff
@@ -38,3 +38,12 @@ export const updateCardConfig = (
...
+  hass: HomeAssistant,
+  cardId: string
+): Promise<string> =>
```
Comment (balloob) [inline src/panels/lovelace/common/data.ts]: What kind of string is returned?

File: src/panels/lovelace/components/hui-card-options.ts (hunk 1 above)
Comment (balloob) [reply src/panels/lovelace/components/hui-card-options.ts]: Yep, disable button when no ID and wrap it in a tooltip saying "Card has no ID"

File: src/panels/lovelace/common/data.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/common/data.ts]: None...

File: src/panels/lovelace/editor/delete-card.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { HomeAssistant } from "../../../types";
+import { LovelaceCardConfig } from "../types";
+
+export async function confDeleteCard(
+  hass: HomeAssistant,
+  cardConfig: LovelaceCardConfig,
+  reloadLovelace: () => void,
+  noIdCallback: () => void
+): Promise<void> {
+  if (!cardConfig) {
```
Comment (balloob) [inline src/panels/lovelace/editor/delete-card.ts]: This check shouldn't be here, it should just not be called.

File: src/panels/lovelace/editor/delete-card.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    return;
+  }
+  if (!cardConfig.id) {
```
Comment (balloob) [inline src/panels/lovelace/editor/delete-card.ts]: Same (but you already noticed that )

File: src/panels/lovelace/components/hui-card-options.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/components/hui-card-options.ts]: Or show the migrate dialog, like this._editCard() does?

**Review by balloob — APPROVED** [review]

**Pull request merged**